spec/controllers/market_models_controller_spec.rb

```diff
@@ -34,4 +34,28 @@
       expect(response).to redirect_to(new_user_session_path)
     end
   end
+
+  describe "#update" do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:market_model) { FactoryGirl.create(:market_model, user: user) }
+
+    before do
+      sign_in(:user, user)
+
+      3.times do
+        testing_ground = FactoryGirl.create(:testing_ground,
+          market_model: market_model)
+
+        FactoryGirl.create(:business_case,
+                           testing_ground: testing_ground, job_id: 1)
+      end
+    end
+
+    it 'should recalculate the business case for all LESs' do
+      patch :update, id: market_model.id, market_model: {
+        name: "test", public: true, interactions: [] }
+
+      expect(BusinessCase.pluck(:job_id).compact).to be_empty
```

Comment:
An explicit check for `expect(...).to eq([nil, nil, nil])` would be better. What if `patch :update` accidentally *deleted* all the business cases? This test would still pass.

It would also be worthwhile to assert that business cases which use a *different* market model are unaffected by the action. If the action instead performed `BusinessCase.update_all(job_id: nil)` the tests would still pass.


```ruby
let(:user) { FactoryGirl.create(:user) }
let(:market_model) { FactoryGirl.create(:market_model, user: user) }
let(:other_market_model) { FactoryGirl.create(:market_model) }

let(:les1) { FactoryGirl.create(:testing_ground, market_model: market_model) }
let(:les2) { FactoryGirl.create(:testing_ground, market_model: other_market_model) }

let(:bc1) { FactoryGirl.create(:business_case, testing_ground: les1, job_id: 1) }
let(:bc2) { FactoryGirl.create(:business_case, testing_ground: les1, job_id: 2) }
let(:bc3) { FactoryGirl.create(:business_case, testing_ground: les2, job_id: 3) }

let(:perform_action) do
  patch :update, id: market_model.id,
    market_model: { name: "test", public: true, interactions: [] }
end

# ...

it 'invalidates job_id on all associated business cases' do
  expect { perform_action }
    .to change { [bc1, bc2].map { |bc| bc.reload.job_id } }
    .from([1, 2]).to([nil, nil])
end

it 'does not invalidate job_id on unrelated business cases' do
  expect { perform_action }.not_to change { bc3.reload.job_id }.from(3)
end
```